api-tests/core/strapi/document-service/relations/no-dp.test.api.ts

```diff
@@ -0,0 +1,210 @@
+      relation: 'oneToMany',
+      target: PRODUCT_UID,
+    },
+  },
+  options: {
+    withDraftAndPublish: false,
+  },
+  displayName: 'Shop',
+  singularName: 'shop',
+  pluralName: 'shops',
+};
+
+describe('Document Service relations', () => {
+  beforeAll(async () => {
+    await builder
+      .addContentTypes([tagModel, productModel, shopModel])
+      .addFixtures('plugin::i18n.locale', [{ name: 'Es', code: 'es' }])
+      .build();
+
+    strapi = await createStrapiInstance();
+    rq = await createAuthRequest({ strapi });
+
+    // PRODUCTS
+    await strapi.db.query(PRODUCT_UID).createMany({
+      data: [
+        { documentId: 'Skate', name: 'Skate-En', locale: 'en', publishedAt: null },
+        { documentId: 'Skate', name: 'Skate-En', locale: 'en', publishedAt: new Date() },
+        { documentId: 'Skate', name: 'Skate-Es', locale: 'es', publishedAt: null },
+        { documentId: 'Candle', name: 'Candle-En', locale: 'en', publishedAt: null },
+        { documentId: 'Candle', name: 'Candle-Es', locale: 'es', publishedAt: null },
+        { documentId: 'Mug', name: 'Mug-En', locale: 'en', publishedAt: null },
+        { documentId: 'Mug', name: 'Mug-Es', locale: 'es', publishedAt: null },
+      ],
+    });
+
+    // TAGS
+    await strapi.db.query(TAG_UID).createMany({
+      data: [
+        { documentId: 'Tag1', name: 'Tag1', publishedAt: null },
+        { documentId: 'Tag2', name: 'Tag2', publishedAt: null },
+        { documentId: 'Tag3', name: 'Tag3', publishedAt: null },
+      ],
+    });
+
+    productDocuments = strapi.documents(PRODUCT_UID);
+    tagDocuments = strapi.documents(TAG_UID);
+    shopDocuments = strapi.documents(SHOP_UID);
+  });
+
+  afterAll(async () => {
+    // Delete all locales that have been created
+    await strapi.db.query('plugin::i18n.locale').deleteMany({ where: { code: { $ne: 'en' } } });
+
+    await strapi.destroy();
+    await builder.cleanup();
+  });
+
+  describe('Non DP (Shop) -> DP (Product)', () => {
+    const xToOneRelations = ['products_ow', 'products_oo', 'products_mo'];
+
+    describe('X to One relation', () => {
+      it('Connect to both draft and publish version by default', async () => {
+        // Create shop targeting a draft product
+        const shop = await shopDocuments.create({
+          data: {
+            name: 'Shop1',
+            products_mo: { documentId: 'Skate', locale: 'en' },
+            products_oo: { documentId: 'Skate', locale: 'en' },
+            products_ow: { documentId: 'Skate', locale: 'en' },
+          },
+          locale: 'en',
+          populate: xToOneRelations,
+        });
+
+        // Products should be connected to both draft and published version
+        xToOneRelations.forEach((relation) => {
+          expect(shop[relation]).toMatchObject([
+            { name: 'Skate-En', locale: 'en', publishedAt: null },
+            { name: 'Skate-En', locale: 'en', publishedAt: expect.any(String) },
+          ]);
+        });
```

Comment:
@alexandrebodin I would like to review the set of tests for relations together 

Specially, I would like to discuss the approach on return values of "oneToOne" relations, if a one to one relation can return a published and draft version, it has to be an array. 

That will be a breaking change, and I wonder if it just would make sense (in xToOne relations) to always return a single relation at a time (draft or publish and pick a default)